### ISSUE TITLE
Some data edition tasks are creating preservation events with insufficient information

### ISSUE THREAD

**Issue by luis100** [keeps/roda#922]
Steps to reproduce:
1. On a representation, create a folder and upload a file to it
2. Move file to representation root
3. Delete folder
4. Delete file

~~The move file event was not found anywhere.~~
The delete file and folder was not associated with the AIP and representation, even on the description, and where on the repository-level events folder, so there is no way to know from where it came.

![roda - catalogue 2](https://cloud.githubusercontent.com/assets/631728/25410203/4718ee50-2a0d-11e7-8fba-65f64f03166b.jpg)

**Comment by luis100** [comment]
The same happens when deleting a representation, there is no information about which AIP it belonged to.

To note that the move and deletion of files and folders should be on the representation preservation metadata, connected with the representation, and the delete of representations should be on the AIP preservation metadata, connected with the AIP. Only the deletion of AIPs should be on the repositories preservation metadata, and should describe the AIP.

**Comment by luis100** [comment]
Move event has not been created due to an error on server due to ServerMessages.properties on the used implementation was not up-to-date.

```
2017-04-26 10:28:59,626 [JobsSystem-io-2-dispatcher-56] ERROR o.r.c.plugins.plugins.PluginHelper - Unexpected exception during 'objectsLogic' execution
java.util.MissingFormatArgumentException: Format specifier '%s'
	at java.util.Formatter.format(Formatter.java:2519)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2940)
	at org.roda.core.common.Messages.getTranslationWithArgs(Messages.java:50)
	at org.roda.core.plugins.plugins.internal.MovePlugin.processFile(MovePlugin.java:214)
	at org.roda.core.plugins.plugins.internal.MovePlugin.access$100(MovePlugin.java:57)
	at org.roda.core.plugins.plugins.internal.MovePlugin$1.process(MovePlugin.java:142)
	at org.roda.core.plugins.plugins.PluginHelper.processObjects(PluginHelper.java:132)
	at org.roda.core.plugins.plugins.internal.MovePlugin.execute(MovePlugin.java:131)
	at org.roda.core.plugins.orchestrate.akka.AkkaWorkerActor.handlePluginExecuteIsReady(AkkaWorkerActor.java:54)
	at org.roda.core.plugins.orchestrate.akka.AkkaWorkerActor.onReceive(AkkaWorkerActor.java:39)
	at akka.actor.UntypedActor$$anonfun$receive$1.applyOrElse(UntypedActor.scala:165)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:496)
	at akka.actor.UntypedActor.aroundReceive(UntypedActor.scala:95)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by nunovieira220** [comment]
It should be solved. Can you confirm it, @luis100 ?

**Comment by luis100** [comment]
The move part works fine for me now.
I changed this issue description to reflect that, the problem now are the delete events location and links to objects.

**Comment by nunovieira220** [comment]
Solved on a0e1c8ec1c233f622c3039a3ab0661313d4c742e. The delete preservation event is still on a repository level but now contains the immediate higher entity as a source identifier. As an example, if a remove a file, RODA creates a repository event with the representation as source. Is it okay, @luis100 ?

**Comment by luis100** [comment]
Define permissions events are also on the wrong place and with insufficient information, it should link to the objects that the permissions were changed to.

![roda - catalogue 6](https://cloud.githubusercontent.com/assets/631728/25495436/fb2b97fe-2b74-11e7-8018-65a6e7904102.jpg)

**Comment by luis100** [comment]
So, up to now, the events that need to be fixed:
- [ ] Delete events
- [ ] Change permissions events

**Comment by luis100** [comment]
@nunovieira220 fix also the update permissions and then I'll test both.

**Comment by nunovieira220** [comment]
Done, please confirm it @luis100

**Comment by luis100** [comment]
- [ ] Permission update event should have a link to the object updated,
- [ ] Permission update event detail message should be say something about this being a permission update! (@jmaferreira please advise)

![roda - catalogue 7](https://cloud.githubusercontent.com/assets/631728/25613727/3488d830-2f27-11e7-8c34-80c6db556549.jpg)

@jmaferreira a question: should an AIP delete pres. event link to the now non-existing AIP? We would support this like the following screenshot (that shows a event of a representation deleted from an AIP which was also deleted)
![roda - catalogue 8](https://cloud.githubusercontent.com/assets/631728/25613988/1606f076-2f28-11e7-98a5-d2febe770895.jpg)

**Comment by jmaferreira** [comment]
If possible be as clear as possible in the detail about the action.

**Comment by nunovieira220** [comment]
Solved on f54f1596b402e4d6c523714f50e4ae5bba4ecc91
